subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -229,13 +236,42 @@ static int tls_release(struct tls_context *tls)
 	return 0;
 }
 
+/* Poll-like function for blocking on TLS sockets. */
```

Comment:
Why "poll-like" instead of the proper poll? Overall, all this stuff adds a lot of complication for unknown benefit.

Reply:
Originally I've implemented it with `poll`, which indeed was simpler. It didn't work-out though as `poll` isn't reentrant (uses static event array), so running two separate sockets on two separate threads simply wasn't possible.